rmf_battery/include/rmf_battery/agv/PowerSystem.hpp

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef RMF_BATTERY__AGV__POWERSYSTEM_HPP
+#define RMF_BATTERY__AGV__POWERSYSTEM_HPP
+
+#include <rmf_utils/impl_ptr.hpp>
+
+namespace rmf_battery {
+namespace agv {
+
+
+class PowerSystem
+{
+public:
+
+  /// Constructor
+  ///
+  /// \param[in] name
+  ///   A string representing the name of this power system
+  ///
+  /// \param[in] nominal_power
+  ///   The rated nominal power consumption in Watts for this power system
+  ///
+  PowerSystem(
+    std::string name,
```

Comment:
Why does a power system need to know its own name? Shouldn't the name be an external label that the power system itself doesn't need to care about?

Reply:
I agree the `PowerSystem` should not require a name. This was previously needed for an older implementation that has changed since. Will get rid of it.

Reply:
Does it make more sense to get rid of the `PowerSystem` class and just supply the `nominal_power` to the constructor of `SimpleDevicePowerSinik` ?

Reply:
If you don't anticipate the need to add more fields to the `PowerSystem` class in the future, then reducing it the way you've described sounds fine to me.

Reply:
Have decided to keep `PowerSystem` for now